### PR TITLE
New Conditions Class

### PR DESCRIPTION
**Pull request by JTrentSmart**
Implements a new, stateful conditions class using the Python dataclass as a base.

This is introduced first as it doesn't depend on or directly modify any other RCAIDE files. It is placed in a 'Missions/Condtions/Conditions.py' directory—note Mission(s), compared to the older 'Mission', for consistency with pluralization of other modules. This subdirectory is used in a follow-up PR creating separate Conditions subclasses for the various types of conditions used within RCAIDE's included missions.

Compared to the prior conditions class, it eliminates the expanded_array micro-class, ones_row functions, and others in favor of type-hinted (NumPy ndarray) attributes, allowing for direct manipulation, expansion, and concatenation/packing of arrays.

It implements a recursive array expansion function across all NumPy arrays within its own and nested child conditions classes, most notably to be used in the (later PR) State class for RCAIDE missions.

This module implements unit tests for basic functionalty.

### CODE REVIEW

**Comment by JTrentSmart** [comment]
Add unpack_array function

**Comment by JTrentSmart** [comment]
Add array multiply, set methods, etc.

File: RCAIDE/Framework/Missions/Conditions/Conditions.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+# ----------------------------------------------------------------------------------------------------------------------
+#  Conditions
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+@dataclass(kw_only=True)
+class Conditions:
+    """
+    A class representing a set of conditions for aerospace simulations.
```
Comment (planes) [inline RCAIDE/Framework/Missions/Conditions/Conditions.py]: change from aerospace to engineering

File: RCAIDE/Framework/Missions/Conditions/Conditions.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    Notes
+    -----
+    - This class is implemented as a dataclass with keyword-only arguments.
+    - It's designed to work with nested Conditions objects and numpy arrays.
+    """
+
+    name: str = 'Conditions'
+
+    number_of_rows: int = 1
+    adjustment_from_parent: int = 0
```
Comment (planes) [inline RCAIDE/Framework/Missions/Conditions/Conditions.py]: I think we discussed this as a group. But I think we should pick a different name. A suggestion, but not one I'm saying needs to be used.

base_row_size_adjustment ?

**Review by planes — COMMENTED** [review]
Let me know if you need assistance completing this PR

File: RCAIDE/Framework/Missions/Conditions/Conditions.py (hunk 1 above)
Comment (JTrentSmart) [reply RCAIDE/Framework/Missions/Conditions/Conditions.py]: Changed

File: RCAIDE/Framework/Missions/Conditions/Conditions.py (hunk 2 above)
Comment (JTrentSmart) [reply RCAIDE/Framework/Missions/Conditions/Conditions.py]: Changed to 'row_size_adjustment'

**Review by planes — APPROVED** [review]
Group reviewed online.

**Pull request merged**